src/sourceMachine.ts

```diff
@@ -286,8 +303,14 @@ export const makeSourceMachine = (params: {
                     assign({
                       sourceRawContent: (ctx, e) => e.code,
                     }),
-                    forwardTo('codeCacheMachine'),
-                    forwardTo('confirmBeforeLeavingMachine'),
+                    pure(() => {
+                      if (!params.isEmbedded) {
```

Comment:
Nit: Maybe we can handle this in a cond? Or a choose? Something more visualisable.